lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3293,29 +3293,14 @@ namespace {
 
       validatePrivateFileIDAttributes(decl);
 
-      // If this module is declared as a C++ module, try to synthesize
-      // conformances to Swift protocols from the Cxx module.
-      auto clangModule = Impl.getClangOwningModule(result->getClangNode());
-      if (!clangModule || requiresCPlusPlus(clangModule)) {
-        if (auto nominalDecl = dyn_cast<NominalTypeDecl>(result)) {
-          conformToCxxIteratorIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxSequenceIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxConvertibleToBoolIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxSetIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxDictionaryIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxPairIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxOptionalIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxVectorIfNeeded(Impl, nominalDecl, decl);
-          conformToCxxSpanIfNeeded(Impl, nominalDecl, decl);
-
-          if (Impl.needsClosureConstructor(decl)) {
-            if (auto closureCtor =
-                    synthesizer.makeClosureConstructor(nominalDecl))
-              nominalDecl->addMember(closureCtor);
-          }
+      if (auto *nominalResult = dyn_cast<NominalTypeDecl>(result)) {
+        deriveAutomaticCxxConformances(Impl, nominalResult, decl);
+
+        if (Impl.needsClosureConstructor(decl)) {
+          if (auto *ctor = synthesizer.makeClosureConstructor(nominalResult))
+            nominalResult->addMember(ctor);
```

Comment:
@egorzhdan this drops the guard for `requiresCPlusPlus()`, do you think that's ok?

I figured the guard was there to prevent `conformToCxx{Itereator,Sequence,ConvertibleToBool}IfNeeded` from pulling in CxxStdlib, but if `decl` is already `std::function` then the user should already need/have that, right?

The other option was to push this into `deriveAutomaticCxxConformances()`, where I can even match against `std::function` in the StringSwitch in `identifyCxxStdTypeByName()`. But I decided against doing so because `synthesizer.makeClosureConstructor()` isn't really related to deriving any protocol conformances, as far as I understand it.